libs/pipelines/api_pipeline.py

```diff
@@ -80,60 +102,118 @@ def run_projections(
 
 def _generate_api_without_ts(projection_result, row, input_dir):
     if projection_result.aggregation_level == AggregationLevel.STATE:
-        generated_data = api.generate_api_for_state_projection_row(
-            row
-        )
+        generated_data = api.generate_api_for_state_projection_row(row)
     elif projection_result.aggregation_level == AggregationLevel.COUNTY:
-        generated_data = api.generate_api_for_county_projection_row(
-            row
-        )
+        generated_data = api.generate_api_for_county_projection_row(row)
     else:
         raise ValueError("Aggregate Level not supported by api generation")
     key_prefix = _get_api_prefix(projection_result.aggregation_level, row)
     generated_key = f"{key_prefix}.{projection_result.intervention.name}"
-    return APIGenerationRow(generated_key, generated_data)
+    return APIOutput(generated_key, generated_data)
 
 
 def _generate_api_with_ts(projection_result, row, input_dir):
     if projection_result.aggregation_level == AggregationLevel.STATE:
-        generated_data = api.generate_api_for_state_timeseries(
+        generated_data = api.generate_state_timeseries(
             row, projection_result.intervention, input_dir
         )
     elif projection_result.aggregation_level == AggregationLevel.COUNTY:
-        generated_data = api.generate_api_for_county_timeseries(
+        generated_data = api.generate_county_timeseries(
             row, projection_result.intervention, input_dir
         )
     else:
         raise ValueError("Aggregate Level not supported by api generation")
     key_prefix = _get_api_prefix(projection_result.aggregation_level, row)
     generated_key = f"{key_prefix}.{projection_result.intervention.name}.timeseries"
-    return APIGenerationRow(generated_key, generated_data)
+    return APIOutput(generated_key, generated_data)
 
 
 def generate_api(
     projection_result: APIPipelineProjectionResult, input_dir: str
-) -> APIGeneration:
+) -> List[APIOutput]:
     """
     pipethrough the rows of the projection
     if it's a county generate the key for counties:
-        /us/counties/{FIPS_CODE}.{INTERVENTION}.json 
-    if it's a state generate the key for states 
+        /us/counties/{FIPS_CODE}.{INTERVENTION}.json
+    if it's a state generate the key for states
         /us/states/{STATE_ABBREV}.{INTERVENTION}.json
     """
     results = []
     for index, row in projection_result.projection_df.iterrows():
         results.append(_generate_api_without_ts(projection_result, row, input_dir))
         results.append(_generate_api_with_ts(projection_result, row, input_dir))
-    return APIGeneration(results)
-
-
-def deploy_results(result: APIGeneration, output: str):
+    return results
+
+
+def build_states_summary(state_data: List[APIOutput], intervention) -> APIOutput:
+    state_summaries = [
+        output.data
+        for output in state_data
+        # Don't love using type here, but CovidActNowStateTimeseries inherits
```

Comment:
is the issue here that we return a list now instead of a tuple? like can't we just keep it a tuple?